pinc/project_holds.inc

```diff
@@ -0,0 +1,46 @@
+<?php
+
+include_once($relPath.'project_events.inc'); // log_project_event
+include_once($relPath.'misc.inc'); // surround_and_join
+
+function remove_holds($projectid, $states)
+{
+    global $pguser;
+
+    $states_str = surround_and_join( $states, "'", "'", ", " );
+    $sql = "
+        DELETE FROM project_holds
+        WHERE projectid='$projectid'
+            AND state in ($states_str)
+    ";
+    try_query($sql);
+    log_project_event( $projectid, $pguser, 'remove_holds', join($states, ' ') );
+}
+
+function add_holds($projectid, $states)
+{
+    global $pguser;
+
+    $values = '';
+    foreach ($states as $state)
+    {
+        if ($values) $values .= ",\n";
+        $values .= "('$projectid', '$state')";
+    }
+    $sql = "
+        INSERT INTO project_holds
+        VALUES $values
+    ";
+    try_query($sql);
+    log_project_event( $projectid, $pguser, 'add_holds', join($states, ' ') );
+}
+
+function try_query($sql)
```

Comment:
Please don't make yet another function to wrap mysqli() calls. If you want to have mysqli throw an exception directly, consider using `mysqli_report(MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT)` before the call in the function instead.